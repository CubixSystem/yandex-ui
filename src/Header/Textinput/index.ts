export * from './_view/Textinput_view_search-arrow';
