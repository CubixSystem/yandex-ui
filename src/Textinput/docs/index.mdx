---
id: Textinput
path: /components/textinput

title: Textinput
description: Однострочное текстовое поле.

links:
 - label: source
   url: https://github.yandex-team.ru/search-interfaces/frontend/tree/master/packages/lego-components/src/Textinput
 - label: startrek
   url: https://search.yandex-team.ru/stsearch?text=Textinput.ts&amp;facet.queue=ISL&amp;facet.type=bug&amp;facet.status=128
 - label: figma
   url: https://www.figma.com/file/CaoHFEO0w6PaE8cVaYFDoq8i/LEGO

tabs:
 - Examples
 - Usage
 - Playground
 - Props
---


