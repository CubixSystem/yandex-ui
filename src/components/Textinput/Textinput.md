# Textinput

<a href="https://github.yandex-team.ru/search-interfaces/frontend/tree/master/packages/lego-components/src/components/Textinput" target="_blank"><img src='https://badger.yandex-team.ru/custom/[Исходники]/[Github
][green]/badge.svg' /></a> <a href="https://search.yandex-team.ru/stsearch?text=Textinput.ts&amp;facet.queue=ISL&amp;facet.type=bug&amp;facet.status=128" target="_blank"><img src='https://badger.yandex-team.ru/custom/[Известные проблемы]/[Startrek][blue]/badge.svg' /></a>

<!-- description:start -->
Однострочное текстовое поле.
<!-- description:end -->

## Пример использования

Конфигурация темы на уровне проекта:

```ts
// src/lib/theme.ts
import { configureRootTheme } from '@yandex-lego/components/Theme'
import { theme } from '@yandex-lego/components/Theme/presets/default'

configureRootTheme({ theme })
```

Использование с нужным набором модификаторов:

```ts
// src/App.ts
import React, { useState } from 'react'
import { compose } from '@bem-react/core'
import {
  Textinput as TextinputDesktop,
  withViewDefault,
  withSizeM,
} from '@yandex-lego/components/Textinput/desktop'

// Композиция из различных модификаторов
const Textinput = compose(withViewDefault, withSizeM)(TextinputDesktop)

const App = () => {
  const [value, setValue] = useState('')

  return (
    <Textinput
      size="m"
      view="default"
      value={value}
      onChange={(event) => setValue(event.target.value)}
    />
  )
}
```

Использование с полным набором модификаторов:

```ts
// src/App.ts
import React, { useState } from 'react'
import { Textinput } from '@yandex-lego/components/Textinput/desktop/bundle'

const App = () => {
  const [value, setValue] = useState('')

  return (
    <Textinput
      size="m"
      view="default"
      value={value}
      onChange={(event) => setValue(event.target.value)}
    />
  )
}
```

## Примеры

### Вид поля

Чтобы изменить вид поля, установите свойство `view` в значение `"default"`.

{{%story::desktop:controls-textinput-desktop--view%}}

### Стилевое оформление

Чтобы изменить стилевое оформление поля, установите свойство `theme` в одно из следующих значений: `"normal"`, `"websearch"`.

{{%story::desktop:controls-textinput-desktop--theme%}}

### Размер поля

Чтобы изменить размер поля, установите свойство `size` в одно из следующих значений: `"m"`, `"s"`.

{{%story::desktop:controls-textinput-desktop--size%}}

### Выравнивание

Чтобы выровнять компоненты по базовой линии, установите свойство `baseline`.

{{%story::desktop:controls-textinput-desktop--baseline%}}

### Тип поля

Чтобы изменить тип поля, установите свойство `type`.

{{%story::desktop:controls-textinput-desktop--type%}}

### Границы поля

Чтобы изменить границы поля, установите свойство `pin`.

{{%story::desktop:controls-textinput-desktop--pin%}}

### Состояние ошибки

Чтобы задать состояние ошибки нужно передать свойство `state` со значением `error`.
Чтобы установить текст ошибки нужно передать значение в свойство `hint`.

{{%story::desktop:controls-textinput-desktop--state%}}

### Крестик для очистки

Чтобы добавить крестик для очистки текстового поля, установите свойство `hasClear`.
Для появления крестика установите не пустое `value` в компоненте (не работает с `defaultValue`).

{{%story::desktop:controls-textinput-desktop--has-clear%}}

| Свойство      | Тип                                                        | Описание                                      |
| ------------- | ---------------------------------------------------------- | --------------------------------------------- |
| hasClear?     | `false \| true`                                            | Наличие крестика для очистки текстового поля. |
| onClearClick? | `(event: MouseEvent<HTMLSpanElement, MouseEvent>) => void` | Обработчик клика по крестику.                 |

## Свойства

<!-- props:start -->
| Свойство      | Тип                                                                                                                                                                                                                                                               | Описание                                                                                                                                                                                           |
| ------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| onBlur?       | `(event: FocusEvent<HTMLElement>) => void`                                                                                                                                                                                                                        | Событие, которое вызывается при потере фокуса компонентом. Например, при клике на другом месте экрана                                                                                              |
| onClick?      | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`                                                                                                                                                                                                            | Событие, которое вызывается при нажатии на компонент                                                                                                                                               |
| onKeyDown?    | `(event: KeyboardEvent<HTMLElement>) => void`                                                                                                                                                                                                                     | Событие, которое вызывается при нажатии клавиш клавиатуры                                                                                                                                          |
| onFocus?      | `(event: FocusEvent<HTMLElement>) => void`                                                                                                                                                                                                                        | Событие, которое возникает при получении компонентом фокуса                                                                                                                                        |
| onMouseDown?  | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`                                                                                                                                                                                                            | Событие по своему действию похоже на `onClick` и возникает в момент нажатия кнопки мыши. `onClick` в каком-то смысле является комбинацией событий `onMouseDown` и `onMouseUp`                      |
| onMouseUp?    | `(event: MouseEvent<HTMLElement, MouseEvent>) => void`                                                                                                                                                                                                            | Событие по своему действию противоположно событию `onMouseDown` и происходит при отпускании кнопки мыши. Курсор должен находится в пределах HTML-элемента, к которому добавлен атрибут `onmouseup` |
| focused?      | `false \| true`                                                                                                                                                                                                                                                   | Состояние фокуса на компоненте                                                                                                                                                                     |
| pressed?      | `false \| true`                                                                                                                                                                                                                                                   | Состояние нажатия на компоненте                                                                                                                                                                    |
| addonAfter?   | `string \| number \| false \| true \| {} \| ReactElement<any, string \| ((props: any) => ReactElement<any, string \| ... \| (new (props: any) => Component<any, any, any>)>) \| (new (props: any) => Component<any, any, any>)> \| ReactNodeArray \| ReactPortal` | Дополнительный контент после контрола                                                                                                                                                              |
| addonBefore?  | `string \| number \| false \| true \| {} \| ReactElement<any, string \| ((props: any) => ReactElement<any, string \| ... \| (new (props: any) => Component<any, any, any>)>) \| (new (props: any) => Component<any, any, any>)> \| ReactNodeArray \| ReactPortal` | Дополнительный контент перед контролом                                                                                                                                                             |
| iconLeft?     | `ReactElement<IIconProps, string \| ((props: any) => ReactElement<any, string \| ... \| (new (props: any) => Component<any, any, any>)>) \| (new (props: any) => Component<any, any, any>)>`                                                                      | Иконка слева от содержимого текстового поля                                                                                                                                                        |
| iconRight?    | `ReactElement<IIconProps, string \| ((props: any) => ReactElement<any, string \| ... \| (new (props: any) => Component<any, any, any>)>) \| (new (props: any) => Component<any, any, any>)>`                                                                      | Иконка справа от содержимого текстового поля                                                                                                                                                       |
| innerRef?     | `(instance: HTMLSpanElement) => void \| RefObject<HTMLSpanElement>`                                                                                                                                                                                               | Ссылка на корневой DOM элемент компонента                                                                                                                                                          |
| style?        | `CSSProperties`                                                                                                                                                                                                                                                   | Пользовательские стили на корневом DOM элементе.                                                                                                                                                   |
| hint?         | `string`                                                                                                                                                                                                                                                          | Текст-подсказка, появляющаяся после компонента. Может иметь различное визуальное оформление в зависимости от свойства `state`.                                                                     |
| state?        | `"error"`                                                                                                                                                                                                                                                         | Визуальное состояние компонента. Может использоваться при проверке формы на корректность.                                                                                                          |
| inputMode?    | `"decimal" \| "numeric" \| "text" \| "url" \| "email" \| "search" \| "tel"`                                                                                                                                                                                       | HTML-атрибут `inputmode`                                                                                                                                                                           |
| controlRef?   | `(instance: HTMLInputElement) => void \| RefObject<HTMLInputElement>`                                                                                                                                                                                             | Ссылка на DOM элемент нативного инпута                                                                                                                                                             |
| autoFocus?    | `false \| true`                                                                                                                                                                                                                                                   | HTML-атрибут `autofocus`                                                                                                                                                                           |
| autoComplete? | `string`                                                                                                                                                                                                                                                          | HTML-атрибут `autocomplete`                                                                                                                                                                        |
| name?         | `string`                                                                                                                                                                                                                                                          | Имя компонента                                                                                                                                                                                     |
| placeholder?  | `string`                                                                                                                                                                                                                                                          | Плейсхолдер                                                                                                                                                                                        |
| value?        | `string`                                                                                                                                                                                                                                                          | Значение контрола                                                                                                                                                                                  |
| defaultValue? | `string`                                                                                                                                                                                                                                                          | Значение по умолчанию контрола                                                                                                                                                                     |
| type?         | `string`                                                                                                                                                                                                                                                          | HTML-атрибут `type`                                                                                                                                                                                |
| disabled?     | `false \| true`                                                                                                                                                                                                                                                   | HTML-атрибут `disabled`                                                                                                                                                                            |
| id?           | `string`                                                                                                                                                                                                                                                          | Уникальный id компонента                                                                                                                                                                           |
| className?    | `string`                                                                                                                                                                                                                                                          | Дополнительный класс                                                                                                                                                                               |
| onChange?     | `(event: ChangeEvent<HTMLInputElement>) => void`                                                                                                                                                                                                                  | Обработчик изменения значения                                                                                                                                                                      |
| onMouseLeave? | `(event: MouseEvent<HTMLInputElement, MouseEvent>) => void`                                                                                                                                                                                                       | Обработчик события `onMouseLeave`                                                                                                                                                                  |
| onMouseEnter? | `(event: MouseEvent<HTMLInputElement, MouseEvent>) => void`                                                                                                                                                                                                       | Обработчик события `onMouseEnter`                                                                                                                                                                  |
<!-- props:end -->
