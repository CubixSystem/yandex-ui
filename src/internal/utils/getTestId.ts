/**
 * Возвращает ID для использования в тестах.
 *
 * @example
 *
 * getTestId('host', testId)
 */
export function getTestId(prefix:string, id?: string): string | undefined {
    if (id === undefined) {
        return undefined;
    }
    return `${id}-${prefix}`;
}
