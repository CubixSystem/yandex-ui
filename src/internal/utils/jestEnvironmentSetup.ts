/**
 * Возвращает DOM-узел для монтирования компонента.
 */
export const getRoot = (): HTMLElement | null => {
    return document.getElementById('root');
};

/**
 * Хелпер, создающий корневой контейнер для монтирования,
 * необходим для того, чтобы избежать ошибок при монтировании в enzyme.
 */
export const jestEnvironmentSetup = (): void => {
    // eslint-disable-next-line no-undef
    beforeEach(() => {
        const node = document.createElement('div');
        node.setAttribute('id', 'root');
        document.body.appendChild(node);
    });

    // eslint-disable-next-line no-undef
    afterEach(() => {
        const node = getRoot();
        if (node !== null) {
            document.body.removeChild(node);
        }
    });
};
