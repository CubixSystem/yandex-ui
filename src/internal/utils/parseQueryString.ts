function parseValue(value: any) {
    if (!isNaN(Number(value)) && (typeof value === 'string' && value.trim() !== '')) {
        value = Number(value);
    } else if (value !== null && (value.toLowerCase() === 'true' || value.toLowerCase() === 'false')) {
        value = value.toLowerCase() === 'true';
    }
    return value;
}

export function parseQueryString<T>(search: string): T {
    const hashes = search.slice(search.indexOf('?') + 1).split('&');
    return hashes.reduce(
        (params, hash) => {
            const split = hash.indexOf('=');
            const key = hash.slice(0, split);
            const value = parseValue(hash.slice(split + 1));
            return Object.assign(params, { [key]: value });
        },
        {} as any,
    );
}
