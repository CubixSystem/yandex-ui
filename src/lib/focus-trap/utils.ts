import { getTabbables } from './tabbable';

const FOCUS_TRAP_GUARD_ATTR = 'focusTrapGuard';

interface ScrollableElement {
    element: HTMLElement;
    scrollTop: number;
    scrollLeft: number;
}

let supportsPreventScrollCached: boolean | null = null;
function supportsPreventScroll() {
    if (supportsPreventScrollCached === null) {
        supportsPreventScrollCached = false;

        try {
            const element = document.createElement('div');

            element.focus({
                get preventScroll() {
                    supportsPreventScrollCached = true;
                    return true;
                },
            });
        } catch (e) {
            // ignore
        }
    }

    return supportsPreventScrollCached;
}

function getScrollableElements(element: HTMLElement): ScrollableElement[] {
    const scrollableElements: ScrollableElement[] = [];
    const rootScrollingElement = document.scrollingElement || document.documentElement;

    let parent = element.parentNode;
    while (parent instanceof HTMLElement && parent !== rootScrollingElement) {
        if (parent.offsetHeight < parent.scrollHeight || parent.offsetWidth < parent.scrollWidth) {
            scrollableElements.push({
                element: parent,
                scrollTop: parent.scrollTop,
                scrollLeft: parent.scrollLeft,
            });
        }
        parent = parent.parentNode;
    }

    if (rootScrollingElement instanceof HTMLElement) {
        scrollableElements.push({
            element: rootScrollingElement,
            scrollTop: rootScrollingElement.scrollTop,
            scrollLeft: rootScrollingElement.scrollLeft,
        });
    }

    return scrollableElements;
}

function restoreScrollPosition(scrollableElements: ScrollableElement[]) {
    for (const { element, scrollTop, scrollLeft } of scrollableElements) {
        element.scrollTop = scrollTop;
        element.scrollLeft = scrollLeft;
    }
}

export function getActiveElement() {
    return document.activeElement as HTMLElement | null;
}

export function isElementInScope(scope: Element, element: Element) {
    return scope.contains(element);
}

export function isFocusFree() {
    return getActiveElement() === document.body;
}

export function createGuard(tabIndex: number) {
    const element = document.createElement('span');
    element.tabIndex = tabIndex;
    Object.assign(element.style, {
        width: '1px',
        height: 0,
        padding: 0,
        overflow: 'hidden',
        position: 'fixed',
        top: '1px',
        left: '1px',
    });
    element.setAttribute('aria-hidden', 'true');
    element.dataset[FOCUS_TRAP_GUARD_ATTR] = 'true';

    return element;
}

export function isGuard(node: HTMLElement) {
    return Boolean(node.dataset[FOCUS_TRAP_GUARD_ATTR]);
}

export function removeElementFromDocument(element: Element) {
    if (!element.parentNode) {
        return;
    }

    element.parentNode.removeChild(element);
}

function focusWithoutScrolling(element: HTMLElement) {
    if (supportsPreventScroll()) {
        element.focus({ preventScroll: true });
    } else {
        const scrollableElements = getScrollableElements(element);
        element.focus();
        restoreScrollPosition(scrollableElements);
    }
}

export function focusElement(element?: HTMLElement | null, preventScroll?: boolean) {
    if (!element) {
        return;
    }

    try {
        if (preventScroll) {
            focusWithoutScrolling(element);
        } else {
            element.focus();
        }
    } catch (e) {
        // ignore
    }
}

export function focusFirstIn(scope: HTMLElement) {
    const [first] = getTabbables(scope, (node) => !isGuard(node));

    focusElement(first);
}

export function getNextTabbableIndex(current: number, count: number, step: number) {
    if (current === -1) {
        return 0;
    }

    return count > 0 ? (count + current + step) % count : -1;
}
