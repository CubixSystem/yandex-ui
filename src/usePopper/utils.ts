import { Placement, BasePlacement, Offsets, top, left, right } from '@popperjs/core';

export function getLayoutRect(element: HTMLElement) {
    return {
        x: element.offsetLeft,
        y: element.offsetTop,
        width: element.offsetWidth,
        height: element.offsetHeight,
    };
}

export default function getBasePlacement(placement: Placement): BasePlacement {
    return placement.split('-')[0] as any;
}

export function distanceAndSkiddingToXY(placement: Placement, offset: [number, number]): Offsets {
    const basePlacement = getBasePlacement(placement);
    const invertDistance = [left, top].indexOf(basePlacement as any) >= 0 ? -1 : 1;

    const skidding = offset[0] || 0;
    const distance = (offset[1] || 0) * invertDistance;

    return [left, right].indexOf(basePlacement as any) >= 0
        ? { x: distance, y: skidding }
        : { x: skidding, y: distance };
}
