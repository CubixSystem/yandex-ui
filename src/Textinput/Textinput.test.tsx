import React, { createRef, ComponentType } from 'react';
import { mount, render } from 'enzyme';
import { ExtractProps } from '@bem-react/core';

import { Textinput as TextinputCommon } from './Textinput';
import { Textinput as TextinputDesktop } from './Textinput@desktop';
import { Textinput as TextinputTouchPad } from './Textinput@touch-pad';
import { Textinput as TextinputTouchPhone } from './Textinput@touch-phone';
import { serverEnvironmentSetup, delay } from '../internal/utils';

const platforms = [
    ['desktop', TextinputDesktop],
    ['touch-pad', TextinputTouchPad],
    ['touch-phone', TextinputTouchPhone],
];

type TextinputProps = ExtractProps<typeof TextinputDesktop>;

describe.each<any>(platforms)('Textinput@%s', (_platform, Textinput: ComponentType<TextinputProps>) => {
    describe('server environment', () => {
        serverEnvironmentSetup();

        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            expect(render(<Textinput className="mix" />)).toMatchSnapshot();
        });
    });

    describe('client environment', () => {
        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            expect(render(<Textinput className="mix" />)).toMatchSnapshot();
        });

        test('должен быть установлен корректный displayName', () => {
            //Так написано, потому что компонент сразу поставляется в обертке.
            expect(TextinputCommon.displayName).toBe('WithControl(Textinput)');
        });

        test('должен устанавливать ссылку на корневой DOM элемент', async () => {
            const innerRef = createRef<HTMLElement>();
            mount(<Textinput innerRef={innerRef} />);
            await delay(100);
            expect(innerRef.current).not.toBe(null);
        });

        test('должен устанавливать ссылку на нативный DOM инпут', async () => {
            const controlRef = createRef<HTMLInputElement>();
            mount(<Textinput controlRef={controlRef} />);
            await delay(100);
            expect(controlRef.current).not.toBe(null);
        });

        test('должен вызывать событие onChange с event при вводе текста в поле', () => {
            const onChange = jest.fn();
            const wrapper = mount(<Textinput onChange={onChange} />);
            wrapper.find('input').simulate('change', { target: { value: 'next-value' } });
            expect(onChange).toHaveBeenCalledTimes(1);
            expect(onChange.mock.calls[0][0].target.value).toBe('next-value');
        });

        test('должен обновлять value в input при обновлении через props', () => {
            const wrapper = mount(<Textinput value="" onChange={() => null} />);
            wrapper.setProps({ value: 'next' });
            expect(wrapper.find('input').prop('value')).toBe('next');
            wrapper.setProps({ value: 'next-value' });
            expect(wrapper.find('input').prop('value')).toBe('next-value');
        });

        test('должен рендерить контент в addonAfter и addonBefore', () => {
            const wrapper = mount(
                <Textinput addonAfter={<i className="after" />} addonBefore={<i className="before" />} />,
            );
            expect(wrapper.find('.after')).toHaveLength(1);
            expect(wrapper.find('.before')).toHaveLength(1);
        });

        test('должен добавлять iconLeft и iconRight', () => {
            const wrapper = mount(<Textinput iconLeft={<i className="left" />} iconRight={<i className="right" />} />);
            expect(wrapper.find('.left')).toHaveLength(1);
            expect(wrapper.find('.right')).toHaveLength(1);
        });

        test('должен добавлять и удалять атрибут disabled при обновлении через props', () => {
            const wrapper = mount(<Textinput />);
            wrapper.setProps({ disabled: true });
            expect(wrapper.find('input').prop('disabled')).toBe(true);
            wrapper.setProps({ disabled: false });
            expect(wrapper.find('input').prop('disabled')).toBe(false);
        });

        test('должен рендерить/удалять элеменет Hint при наличии свойства', () => {
            const wrapper = mount(<Textinput hint="error" />);
            expect(wrapper.find('.Textinput-Hint')).toHaveLength(1);
            wrapper.setProps({ hint: '' });
            wrapper.find('.Textinput-Hint').simulate('animationend');
            expect(wrapper.find('.Textinput-Hint')).toHaveLength(0);
        });
    });
});
