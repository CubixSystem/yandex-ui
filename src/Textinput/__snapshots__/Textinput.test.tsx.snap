// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textinput@desktop client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textinput mix"
>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="Textinput-Control"
  />
  <span
    class="Textinput-Box"
  />
</span>
`;

exports[`Textinput@desktop server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textinput mix"
>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="Textinput-Control"
  />
  <span
    class="Textinput-Box"
  />
</span>
`;

exports[`Textinput@touch-pad client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textinput mix"
>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="Textinput-Control"
  />
  <span
    class="Textinput-Box"
  />
</span>
`;

exports[`Textinput@touch-pad server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textinput mix"
>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="Textinput-Control"
  />
  <span
    class="Textinput-Box"
  />
</span>
`;

exports[`Textinput@touch-phone client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textinput mix"
>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="Textinput-Control"
  />
  <span
    class="Textinput-Box"
  />
</span>
`;

exports[`Textinput@touch-phone server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Textinput mix"
>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="Textinput-Control"
  />
  <span
    class="Textinput-Box"
  />
</span>
`;
